### ISSUE TITLE
Convert GlobalErrorBoundary to a functional component to reduce the i18n complexity

### ISSUE THREAD

**Issue by rszwajko** [oVirt/ovirt-web-ui#1409]
Feature enhancement request:

Follow-up issue after #1396 

With PR #1396 the complexity of GlobalErrorBoundary increased to the point that justifies refactoring - rewriting as a functional component.

**Comment by rszwajko** [comment]
As per React [documentation](https://reactjs.org/docs/hooks-faq.html#do-hooks-cover-all-use-cases-for-classes) there is no functional replacement for `componentDidCatch` so it's not possible to re-write error boundary functionality. We could try extracting only i18n part but I don't see a clean way as  GlobalErrorBoundary requires access to messages and is(should be?) the top level component.

@sjd78 
Do you see another way to refactor this part?

**Comment by sjd78** [comment]
After looking at it a bit, I'd say that `GlobalErrorBoundary` itself keeps all the error trapping as-is.

Then the refactoring would be to take bits out the `render()` function and spin it off to a functional component in the same file.  Something like `GlobalErrorPage` maybe.  Then that functional component should be the only thing that needs the msg context.

**Comment by sjd78** [comment]
> After looking at it a bit, I'd say that `GlobalErrorBoundary` itself keeps all the error trapping as-is.
> 
> Then the refactoring would be to take out the `render()` function and spin it off to a functional component in the same file. Something like `GlobalErrorPage` maybe. Then that functional component should be the only thing that needs the msg context.

At this point, no need to make any changes.  Leaving the component as-is is acceptable.
